### PR TITLE
some functionality needed to replace the old field transform:

### PR DESCRIPTION
**Pull request by martinvandriel**
- follow symlinks (Data dir on /scratch e.g.)
 - allow writing into the same dir (from axisem_output to ordered_output)

### CODE REVIEW

File: instaseis/scripts/repack_db.py [hunk 1]
```diff
@@ -490,7 +490,11 @@ def repack_database(input_folder, output_folder, contiguous,
 
     assert found_filenames, "No files named `ordered_output.nc4` found."
 
-    os.makedirs(output_folder)
+    if input_folder == output_folder:
+        if "ordered_output.nc4" in found_filenames:
```
Comment (krischer) [inline instaseis/scripts/repack_db.py]: `found_filenames` contains the full paths so this conditional will likely never be true.

This should work: 

```python
if "ordered_output.nc4" in [os.path.basename(_i) for _i in found_filenames]:
   ...
```

File: instaseis/scripts/repack_db.py [hunk 2]
```diff
@@ -490,7 +490,11 @@ def repack_database(input_folder, output_folder, contiguous,
 
     assert found_filenames, "No files named `ordered_output.nc4` found."
 
-    os.makedirs(output_folder)
+    if input_folder == output_folder:
```
Comment (krischer) [inline instaseis/scripts/repack_db.py]: A minor nitpick and a bit of a constructed case but why not catch it: `input_folder` and `output_folder` might be different strings but refer to the same folder. Using

```python
input_folder = os.path.normpath(os.path.realpath(input_folder))
output_folder = os.path.normpath(os.path.realpath(output_folder))
if input_folder == output_folder:
    ...
```

would solve this.

Comment (martinvandriel) [inline instaseis/scripts/repack_db.py]: the variables input_folder and output_folder are used afterwards, is it save to overwrite them here? I guess so...

**Comment by krischer** [comment]
Manually merged into the master (not sure why github does not auto-close this) together with a numpy compatibility fix.

**Pull request closed without merging**